bind/files/zone.jinja

```diff
@@ -11,17 +12,22 @@
 {%- set zone_expiry = soa['expiry'] if soa['expiry'] is defined else '2w' %}
 {%- set zone_nxdomain = soa['nxdomain'] if soa['nxdomain'] is defined else '1m' %}
 
+{% if not serial_auto or include %}
 {% if soa['ttl'] is defined %}
 $TTL {{ soa['ttl'] }}
 {% endif %}
 @ {{ zone_class }} SOA {{ zone_ns }} {{ zone_contact }} (
-    {{ zone_serial }} ; serial
+    {{ salt['dnsutil.serial'](zone) if serial_auto else zone_serial }} ; serial
```

Comment:
Here, it's not comparing if `serial_auto == 'auto'` but only checking if it is set. As it is always set, `salt['dnsutil.serial'](zone)` will be always used, and `zone_serial` won't.